pyInterface/module/utils/_binning.py

```diff
@@ -61,15 +63,94 @@ def __ge__(self, other):
 
 
 	def __str__(self):
-		retval = ""
-		retval += "multiBin: { "
+		retval = "multiBin: " + self.__repr__()
+		return retval
+
+	def __repr__(self):
+		retVariables = []
 		for key in sorted(self.boundaries.keys()):
-			retval += "\"" + key + "\": (" + str(self.boundaries[key][0]) + ", " + str(self.boundaries[key][1]) + "), "
-		retval = retval[:-2] + " }"
+			retVariables.append( "\"" + key + "\": (" + str(self.boundaries[key][0]) + ", " + str(self.boundaries[key][1]) + ")")
+		retval = "{ "
+		retval += ", ".join(retVariables)
+		retval += " }"
 		return retval
 
+	def __hash__(self):
+		keys = tuple(self.boundaries.keys())
```

Comment:
Is it ensured that `keys` returns a sorted list? My understanding of the Python reference is that it is not, so we should probably do it ourselves, i.e., use `tuple(self.variables())`